src/engine/source/_builder/asset.hpp

```diff
@@ -0,0 +1,128 @@
+#ifndef _ASSET_H
+#define _ASSET_H
+
+#include <string>
+#include <unordered_set>
+
+#include <fmt/format.h>
+
+#include "_builder/expression.hpp"
+#include "_builder/json.hpp"
+#include "_builder/registry.hpp"
+
+class Asset
+{
+public:
+    enum class Type
```

Comment:
is there a way in which we don't duplicate this enum? there's another with env in Builder.hpp